.github/workflows/wheels.yml

```diff
@@ -242,12 +242,185 @@ jobs:
+          echo ""
+
+          HAS_ERRORS=0
+
+          for tarball in dist/*.tar.gz; do
+            if [ ! -f "$tarball" ]; then
+              echo "⚠️  No source distribution found in dist/"
+              continue
+            fi
+
+            BASENAME=$(basename "$tarball")
+            VERIFY_FILE="dist/${BASENAME%.tar.gz}.verify.txt"
+
+            echo "==> Verifying: $BASENAME"
+            echo ""
+
+            # Create verification report
+            {
+              echo "========================================================================"
+              echo "Source Distribution Integrity Verification Report"
+              echo "========================================================================"
+              echo ""
+              echo "Verification performed at origin (wheels workflow)"
+              echo "This report ensures artifact integrity through the entire pipeline."
+              echo ""
+              echo "------------------------------------------------------------------------"
+              echo "Metadata"
+              echo "------------------------------------------------------------------------"
+              echo "Filename:      $BASENAME"
+              echo "Timestamp:     $(date -u '+%Y-%m-%d %H:%M:%S UTC')"
+              echo "Workflow:      ${{ github.workflow }}"
+              echo "Run ID:        ${{ github.run_id }}"
+              echo "Runner OS:     ${{ runner.os }}"
+              echo "Runner Arch:   ${{ runner.arch }}"
+              echo "Commit SHA:    ${{ github.sha }}"
+              echo "Ref:           ${{ github.ref }}"
+              echo ""
+
+              echo "------------------------------------------------------------------------"
+              echo "File Properties"
+              echo "------------------------------------------------------------------------"
+              ls -lh "$tarball"
+              echo ""
+              echo "File size (bytes): $(stat -c%s "$tarball" 2>/dev/null || stat -f%z "$tarball")"
+              echo ""
+
+              echo "------------------------------------------------------------------------"
+              echo "Cryptographic Fingerprint (SHA256)"
+              echo "------------------------------------------------------------------------"
+              openssl sha256 "$tarball"
+              echo ""
+
+              echo "------------------------------------------------------------------------"
+              echo "Gzip Integrity Test"
+              echo "------------------------------------------------------------------------"
+              if gzip -tv "$tarball" 2>&1; then
+                GZIP_EXIT=$?
+                echo ""
+                echo "Gzip exit code: $GZIP_EXIT"
+                if [ $GZIP_EXIT -eq 0 ]; then
+                  echo "Result: ✅ PASS (no trailing garbage)"
+                else
+                  echo "Result: ❌ FAIL (exit code $GZIP_EXIT)"
```

Comment:
The failure part here is unreachable. If you're in this branch of the parent `if`, it has to be `0`.